static/panes/selectiondag-view.js

```diff
@@ -0,0 +1,567 @@
+        for (i = 0; i < nodes.length; i++) {
+            var id = nodes[i].id;
+            if ( !contains('#'+id, selectorTextArr) )
+                selectorTextArr.push( '#'+id );
+
+            var classes = nodes[i].classList;
+            for (c = 0; c < classes.length; c++)
+                if ( !contains('.'+classes[c], selectorTextArr) )
+                    selectorTextArr.push( '.'+classes[c] );
+        }
+
+        // Extract CSS Rules
+        var extractedCSSText = "";
+        for (i = 0; i < document.styleSheets.length; i++) {
+            var s = document.styleSheets[i];
+
+            try {
+                if(!s.cssRules) continue;
+            } catch( e ) {
+                if(e.name !== 'SecurityError') throw e; // for Firefox
+                continue;
+            }
+
+            var cssRules = s.cssRules;
+            for (var r = 0; r < cssRules.length; r++) {
+                if ( contains( cssRules[r].selectorText, selectorTextArr ) )
+                    extractedCSSText += cssRules[r].cssText;
+            }
+        }
+
+
+        return extractedCSSText;
+
+        function contains(str,arr) {
+            return arr.indexOf(str) !== -1;
+        }
+
+    }
+
+    function appendCSS( cssText, element ) {
+        var styleElement = document.createElement("style");
+        styleElement.setAttribute("type","text/css");
+        styleElement.innerHTML = cssText;
+        var refNode = element.hasChildNodes() ? element.children[0] : null;
+        element.insertBefore( styleElement, refNode );
+    }
+};
+
+SelectionDAG.prototype.saveDAGAsSVG = function () {
+    try {
+        var doctype = 
+            '<?xml version="1.0" standalone="no"?>' + 
+            '<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" ' +
+            '[<!ENTITY nbsp "&#160;">]>';
+        var name = this.currentFunc + "-" + this.currentDag;
+        var graphSvg = this.placeholder.select("svg");
+        var svgContents = graphSvg 
+            .attr("title", name)
+            .attr("version", 1.1)
+            .attr("xmlns", "http://www.w3.org/2000/svg")
+            .attr("xmlns:xhtml", "http://www.w3.org/1999/xhtml")
+            .attr("xmlns:xlink", "http-www.w3.org/1999/xlink")
+            .node().parentNode.innerHTML;
+
+        saveAs(new Blob(
+            [doctype + svgContents],
+            {type: "image/svg+xml"}),
+        name + ".svg");
+        return true;
+    } catch (e) {
+        this.alertSystem.notify('Error while saving your selection DAG. Use the clipboard instead.', {
+            group: "savelocalerror",
+            alertClass: "notification-error",
+            dismissTime: 5000
+        });
+        return false;
+    }
+};
+
+SelectionDAG.prototype.saveDAGAsPNG = function () {
+    try {
+        var name = this.currentFunc + "-" + this.currentDag;
+        var graphSvg = this.placeholder.select("svg");
+        
+        // let svgContents = graphSvg
```

Comment:
My usual "probably this is WIP but would be cool to clean this up when you're ready" :)